Release notes for 3.5.0 (#15)
ocient/pyocient_release: README.md

# Pyocient Release Notes

## 3.5.0

> - Added support for executable SQL files via shebang lines in the command-line interface.

Reviewer: ```suggestion
- Added support for executable SQL files using shebang lines in the command-line interface (CLI).
```

- Added `expect error` and `expect rowcount` commands to the CLI.

> - Added enhanced help functionality to the command-line interface.

Reviewer: ```suggestion
- Added enhanced help functionality to the CLI.
```

- Added nanosecond precision support for timestamp data types.

> - Added server-side `DESCRIBE TABLE/VIEW` command support with fallback to client-side describe for older server versions.

Reviewer: ```suggestion
- Added server-side `DESCRIBE TABLE` and `DESCRIBE VIEW` command support with fallback to the client-side `DESCRIBE` command for older server versions.
```

- Added connection timeout support to prevent hanging connections.
- Added configurable redirect URIs per OIDC integration for SSO functionality.

> - Fixed an infinite loop issue when fetch size was set to 0.

Reviewer: ```suggestion
- Fixed an infinite loop issue when fetch size is set to 0.
```

- Improved parameter expansion handling and error messaging.
- Improved Unicode character formatting in JSON output and CLI display.
- Updated protobuf dependency to version 5.29.2 or higher.
- Reduced verbosity for connection-type logging messages.
- Added support for preserving schema and service class settings after reconnection.

> - Improved error handling for invalid DSNs in the command-line interface.

Reviewer: ```suggestion
- Improved error handling for invalid DSNs in the CLI.
```

> - Removed deprecated AI text-to-SQL functionality.

Reviewer: Please remove this release note as this was internal work that was never exposed externally.
